### DISCUSSION TITLE
my ps2 controller disconnects

### DISCUSSION THREAD

**Discussion by khtp** [darthcloud/BlueRetro#248] [Q&A]
I have problems with my esp32 configured with ps2, in ps2 fat model 30000x it has a time limiter, that is, if I play for 1 hour the controller disconnects and I have to restart esp32 (turn off and on) it already works but I have to carry out this process every hour , it only happens to me with ps2 on other consoles it has not happened to me

**Comment by darthcloud** [comment]
You can try to get the log from the ESP32:
https://github.com/darthcloud/BlueRetro/wiki/Getting-BlueRetro-debug-logs-via-Serial-port-Windows-10

Connect the PC and leave the putty log running the whole session until the issue happen, then copy the full log to a txt file and attach here.

The PC will power the ESP, so disconnect any power pin from console (either 3.3v or 5V) keep the GND connected between esp and console.

**Reply by khtp** [reply]
If I have pin d5, io26, io27 at 3.3v that are bridged, it does not affect when using putty, that is, if I connect the esp32 to the pc and these pins are soldered or does it not matter?

**Comment by khtp** [comment]
[log2esp32.txt](https://github.com/darthcloud/BlueRetro/files/8440057/log2esp32.txt)
[log3esp32.txt](https://github.com/darthcloud/BlueRetro/files/8440058/log3esp32.txt)
[logesp32.txt](https://github.com/darthcloud/BlueRetro/files/8440059/logesp32.txt)
[logfinalesp32.txt](https://github.com/darthcloud/BlueRetro/files/8440060/logfinalesp32.txt)
The same problems happen to me 10 minutes that I don't move anything and it disconnects, the last file I sent you I can't send more because I cause a loop because I send an error 0x.79 or something like that I have to disconnect and connect for it to work again

**Reply by khtp** [reply]
detail 2 that I noticed, if there is another bluethooht in my environment that is in the middle between esp32 and the controller, the controller loses connection and starts to restart the game

**Comment by darthcloud** [comment]
I see you use Xbox X|S controller which is BLE, try if you can with Classic BT controller to see if it's better like PS4 or PS5.

For the cmd error which game are you using?

Btw I know when you reply here, no need to spam me on twitter DMs. I offer support only through github. This is a free project on my spare time. I'll answer when I have time available / feel like it. Ultimately if it doesn't work and you cant wait, use a wired controller.

**Reply by khtp** [reply]
the game is called TOCA Race Driver 3 for ps2, I don't know what the Classic BT controller is, I don't have a ps4 or ps5 controller, I have a wii u pro controller, they are the only bluethooht controllers I have, I'm using the ps2 fat sphc 30,000 USA on ps2 slim I haven't tried but it should work the same I imagine (I don't know why I haven't tried it) the controller is from an xbox one x and it has firmware 5.1 I mean its latest version of this controller I don't have the version of series x/ s

**Reply by khtp** [reply]
ok I understand don't worry, I can wait without any problem I just mention in case someone else may have these problems with the xbox one x controller I think it's second generation, I don't have an s/x series controller, I only have this problem on ps2, now I have tried to play on a wii configured with a gamecube controller and the xbox one x controller works without problems, it supports more than 1 hour, the game TOCA RACE DRIVER 3 for ps2 is being played via the network on a router with a Wi-Fi port, could it be that include in the operation of esp32 or is it close to other bluethooht devices such as the pc and the router for example or does it have nothing to do with it?

**Reply by darthcloud** [reply]
I haven't noticed myself when updating my own Xbox One controller a while back, but the latest software update change controller from a BT classic device into a BLE one.

So this is why it look like an X|S controller in the logs.

I think your issues probably same thing as #102 or #150

But no fix yet.

**Reply by khtp** [reply]
And what do you recommend I do for now while I wait? Do I update to 1.5.1 or download an old update?

Question 2 for ps2 in the programmer the option: "no change to bin" should I leave the tick or remove it?

**Reply by darthcloud** [reply]
All existing FW will exhibit same behavior.

Leave "no change to bin" tick.

**Comment by darthcloud** [comment]
I played two hour on my PS2 FAT SCPH-30001 with BlueRetro v1.6.1 on an official ESP-DevKit-C rev E using an Xbox X S controller with FW 5.13.3143.0.

No issues at all.

Issues you might have:
* Not using official devkitc (Get it on [Digikey](https://www.digikey.com/en/products/detail/espressif-systems/ESP32-DEVKITC-32E/12091810?s=N4IgTCBcDaIKYGcAOBmMBaAJnAbgawEsAXAY3TThAF0BfIA) or [Mouser](https://www.mouser.com/ProductDetail/Espressif-Systems/ESP32-DevKitC-32E?qs=sGAEpiMZZMv0NwlthflBi3XYxq%252BTapwyHnR2%2FrAf4bY%3D))
* ESP Power wiring not using controller port GND and 8V
* Issue with that specific console

In any case I doubt it got anything to do with the Xbox controller, the error you get "controller disconnect" in games relate to the wired part of BlueRetro, a wireless disconnection wouldn't produce that error as BlueRetro emulate the PS controller regardless of BT connection.

Sorry I can't help more than that.

![PXL_20220527_140021597](https://user-images.githubusercontent.com/3744056/170719106-b4f81900-d478-408e-9b77-b519df50bf3d.jpg)

**Reply by delislej** [reply]
tested this myself with a ESP-DevKit-C rev E, worked just fine after 4 hours of shadow hearts on my phat ps2, series S controller.
